expand facility from string (#358)

jagw-mobica opened the pull request:
Modify stumpless_get_facility_enum that the given string matches the end of the current symbol names without the STUMPLESS_FACILITY_ prefix. Exception: 'security' can be treated as identical to 'auth' and is returning STUMPLESS_FACILITY_AUTH.
This is intended to support CLI and other string-based applications.

jagw-mobica committed:
expand facility from string (#358)

Modify stumpless_get_facility_enum that the given string matches
the end of the current symbol names without the STUMPLESS_FACILITY_ prefix.
Exception: 'security' can be treated as identical to 'auth' and
is returning STUMPLESS_FACILITY_AUTH.
This is intended to support CLI and other string-based applications.

Signed-off-by: Jaroslaw Gawin <jaroslaw.gawin@mobica.com>

jagw-mobica commented:
I had to correct the logic in condition (file: priority.c). Now it should works.
Additionally I added test cases for this condition.

goatshriek commented on test/function/facility.cpp:
```diff
@@ -53,11 +57,59 @@ namespace {
+    int result;
+
+    result = stumpless_get_facility_enum( "user" );
+    EXPECT_EQ( result, 8 );
```
For each of these comparisons, use the enum symbol (e.g. `STUMPLESS_FACILITY_USER`) instead of the hardcoded number.

jagw-mobica replied:
Good point, done.

goatshriek reviewed (changes requested):
Thank you very much for putting this together! Just one adjustment for you to make in order for the tests to be more readable.

jagw-mobica committed:
expand severity from string (#359)

Modify stumpless_get_severity_enum that the given string matches
the end of the current symbol names without the STUMPLESS_SEVERITY_ prefix.
This is intended to support CLI and other string-based applications.

Signed-off-by: Jaroslaw Gawin <jaroslaw.gawin@mobica.com>

goatshriek commented on src/priority.c:
```diff
@@ -43,8 +38,7 @@ stumpless_prival_from_string( const char *string ) {
   size_t pre_len;
-  const char pre_facility[] = "STUMPLESS_FACILITY_";
-  const char pre_severity[] = "STUMPLESS_SEVERITY_";
+//  const char pre_severity[] = "STUMPLESS_SEVERITY_";
```
remove commented out code

jagw-mobica replied:
done

goatshriek commented on src/priority.c:
```diff
@@ -94,18 +84,24 @@ stumpless_prival_from_string( const char *string ) {
   // Calculate the severity length
   len = slen - ++len;
-  pre_len = sizeof(pre_severity) - 1;
+//  pre_len = sizeof(pre_severity) - 1;
```
remove commented out code

jagw-mobica replied:
done

goatshriek commented on test/function/priority.cpp:
```diff
@@ -35,6 +35,20 @@ namespace {
+    int result;
+
+    result = stumpless_prival_from_string( "192" );
+    EXPECT_EQ( result, -1 );
```
Also call `EXPECT_ERROR_ID_EQ` after these and make sure that the error is set to the appropriate type.

jagw-mobica replied:
done

goatshriek commented on test/function/severity.cpp:
```diff
@@ -53,11 +57,52 @@ namespace {
+    int result;
+
+    result = stumpless_get_severity_enum( "emerg" );
+    EXPECT_EQ( result, STUMPLESS_SEVERITY_EMERG );
```
include `EXPECT_NO_ERROR` after each of these successful calls to make sure that an error was not left around by mistake.

jagw-mobica replied:
done

goatshriek commented on src/priority.c:
```diff
@@ -94,18 +84,24 @@ stumpless_prival_from_string( const char *string ) {
     return -1;
   }
+
+//  param = alloc_mem( len + 1 + pre_len );
```
remove these blocks of commented code as well

jagw-mobica replied:
done

goatshriek reviewed (changes requested):
Thank you for putting this together, it is mostly ready! Only a few adjustments for you to make.
